src/js/select2/data/ajax.js

```diff
@@ -110,5 +110,19 @@ define([
     }
   };
 
+  AjaxAdapter.prototype.unselect = function (data) {
+    var self = this;
+
+    // Remove matching option elements from the DOM
+    this.current(function (currentData) {
```

Comment:
This feels like a misuse of the `unselect` event -- because we're ignoring the `data` argument -- which ideally would provide all the information we need -- and instead accessing the control's currently-selected items to unselect everything.

Reply:
Resolved: moved the cache invalidation to the `select` function on the AJAX adapter instead.